tools/imagepuller-benchmark/main.go

```diff
@@ -0,0 +1,425 @@
+	if err != nil {
+		return 0, err
+	}
+	return usage.Used, nil
+}
+
+func extractName(name string) string {
+	at := strings.Index(name, "@")
+	if at == -1 {
+		return ""
+	}
+	slash := strings.LastIndex(name[:at], "/")
+	return name[slash+1 : at]
+}
+
+func cleanup(storagePath string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "findmnt", "-rn", "-o", "TARGET")
+	output, err := cmd.Output()
+
+	var exitErr *exec.ExitError
+	if errors.As(err, &exitErr) {
+		return fmt.Errorf("findmnt returned non-zero exit code, stderr is: %s", string(exitErr.Stderr))
+	} else if err != nil {
+		return fmt.Errorf("failed to execute findmnt: %w", err)
+	}
+
+	var mountpoints []string
+	scanner := bufio.NewScanner(strings.NewReader(string(output)))
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, storagePath) {
+			mountpoints = append(mountpoints, line)
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return fmt.Errorf("reading findmnt output: %w", err)
+	}
+
+	sort.Slice(mountpoints, func(i, j int) bool {
+		return strings.Count(mountpoints[i], "/") > strings.Count(mountpoints[j], "/")
+	})
+
+	for _, mp := range mountpoints {
+		if err := unix.Unmount(mp, 0); err != nil {
+			return fmt.Errorf("unmounting %s: %w", mp, err)
+		}
+	}
+
+	if err := os.RemoveAll(storagePath); err != nil {
+		return fmt.Errorf("removing directory %s: %w", storagePath, err)
+	}
+	return nil
+}
+
+func findChildPid(ctx context.Context, ppid int) (int, error) {
+	out, err := exec.CommandContext(ctx, "ps", "-o", "pid=", "--ppid", fmt.Sprint(ppid)).Output()
+
+	var exitErr *exec.ExitError
+	if errors.As(err, &exitErr) {
+		return 0, fmt.Errorf("ps returned non-zero exit code, stderr is: %s", string(exitErr.Stderr))
+	} else if err != nil {
+		return 0, fmt.Errorf("failed to execute ps: %w", err)
+	}
+
+	lines := strings.SplitSeq(string(out), "\n")
+	for line := range lines {
+		line = strings.TrimSpace(line)
+		if line == "" {
+			continue
+		}
+		pid, err := strconv.Atoi(line)
+		if err != nil {
+			continue
+		}
+		return pid, nil
+	}
+	return 0, fmt.Errorf("no child found for PID %d", ppid)
+}
+
+func startServerWithMemoryTracking(ctx context.Context, serverPath string, args ...string) (func() (int, error), int, error) {
+	var stderr bytes.Buffer
+	var cmd *exec.Cmd
+
+	timeCmd := exec.CommandContext(ctx, "which", "time")
+	out, err := timeCmd.Output()
+
+	var exitErr *exec.ExitError
+	if errors.As(err, &exitErr) {
+		return nil, 0, fmt.Errorf("which returned non-zero exit code, stderr is: %s", string(exitErr.Stderr))
+	} else if err != nil {
+		return nil, 0, fmt.Errorf("failed to execute which: %w", err)
+	}
+
+	timePath := strings.TrimSpace(string(out))
+
+	argsFull := append([]string{"-v", serverPath}, args...)
+	cmd = exec.CommandContext(ctx, timePath, argsFull...)
+	cmd.Stdout = nil
+	cmd.Stderr = &stderr
+
+	if err := cmd.Start(); err != nil {
```

Comment:
```suggestion
	cmd.Stderr = &stderr
	if err := cmd.Start(); err != nil {
```
never empty line between a command invocation and its error handling